ducktape/src/test/scala/io/github/arainko/ducktape/issues/Issue38Spec.scala

```diff
@@ -0,0 +1,46 @@
+package io.github.arainko.ducktape.issues
+
+import io.github.arainko.ducktape.*
+import io.github.arainko.ducktape.internal.macros.*
+
+// https://github.com/arainko/ducktape/issues/38
+class Issue38Spec extends DucktapeSuite {
+  final case class TestClass(str: String, int: Int)
+
+  final case class TestClassWithAdditionalList[A](str: String, int: Int, additionalArg: List[A] = Nil)
+
+  final case class TestClassWithMandatoryAdditionalList(str: String, int: Int, additionalArg: List[String])
+
+  test("derive a transformer for case classes with default values if configured") {
```

Comment:
So far so good, I'd also like to see what happens when `A =:!= String` because I feel that there's a failure state hidden here. 

To circumvent that you might need to do a subtype check (`TypeRepr` of the default value <:< `TypeRepr` of the field) at the place where you construct a tree from `Field.default` and only allow it proceed when the  check passes

Reply:
Added the test and new failure. Good thinking!
I didn't even know the compiler allows creating a class with a generic parameter that's not a supertype of the given default value. Seems like something the compiler should do.

Reply:
Yeah, that's the weird thing about defaults that I was worried about